### PR TITLE
Add test and build workflow

### PR DESCRIPTION
**Pull request by shtrom**
This PR
* Adds a Makefile allowing to drive building and tests
* Enables the unit, and integration against the Data API, test suite
* Add Travis CI integration against a number of JDKs

### CODE REVIEW

**Comment by shtrom** [comment]
@klauste Care to have a look? It's just moving things about, but now we have this: https://travis-ci.org/Learnosity/learnosity-sdk-java

File: src/test/java/learnositysdk/request/DataApiIT.java [hunk 1]
```diff
@@ -0,0 +1,193 @@
...
+		}
+
+	public void testGetItemsEmptyRemote()
+			throws java.lang.Exception
+		{
+			String endpoint = baseUrl + "/itembank/items";
+			System.out.println("Testing Data API call to " + endpoint + " with Remote object");
+
+			dataApi = new DataApi(endpoint, securityMap, consumerSecret);
+			response = dataApi.requestJSONObject();
```
Comment (klauste) [inline src/test/java/learnositysdk/request/DataApiIT.java]: You could dry up the code a bit if you put the 5 lines above in their own function. They occur in almost all the tests (with additional request and and request type ('set' or 'get').

File: src/test/java/learnositysdk/request/InitTest.java [hunk 2]
```diff
@@ -0,0 +1,386 @@
...
+		securityObj = new JSONObject();
+		securityObj.put("consumer_key", consumerKey);
+		securityObj.put("user_id", "12345678");
+		securityObj.put("timestamp", "20140612-0438");
+
+		request = new JSONObject ();
+	}
+
+	public void testInitGenerate()
+			throws java.lang.Exception
```
Comment (klauste) [inline src/test/java/learnositysdk/request/InitTest.java]: indentation is off

File: src/test/java/learnositysdk/request/InitTest.java [hunk 3]
```diff
@@ -0,0 +1,386 @@
...
+
+			securityObj.put("domain","demos.learnosity.com");
+
+			init = new Init("assess", securityObj, consumerSecret, request);
+
+			JSONObject signedRequest = new JSONObject(init.generate());
+			assertEquals("Errors in the Assess initialisation",
+					"Demo3",
+					signedRequest.getJSONArray("items").getJSONObject(0).get("reference")
+				    );
```
Comment (klauste) [inline src/test/java/learnositysdk/request/InitTest.java]: indentation

File: src/test/java/learnositysdk/request/DataApiIT.java (hunk 1 above)
Comment (shtrom) [reply src/test/java/learnositysdk/request/DataApiIT.java]: Done, though it's not a big win.

**Comment by shtrom** [comment]
@klauste care for another look?

**Pull request merged**